server/server.go

```diff
@@ -4584,18 +4584,16 @@ func (s *StanServer) addSubscription(ss *subStore, sub *subState) error {
 }
 
 // updateDurable adds back `sub` to the client and updates the store.
-// No lock is needed for `sub` since it has just been created.
-func (s *StanServer) updateDurable(ss *subStore, sub *subState) error {
-	// Reset the hasFailedHB boolean since it may have been set
-	// if the client previously crashed and server set this
-	// flag to its subs.
-	sub.hasFailedHB = false
```

Comment:
This change is not related to the `initialized` reset issue, but I moved that because with the updated test, I got a DATA race report. The actual fix for issue 930 is in processSub() by setting `initialized=false` for a recovered durable subscription.